code/game/objects/structures/oxygen.dm

```diff
@@ -0,0 +1,157 @@
+/obj/structure/oxygen_cabinet

```

Comment:
instead of making a bunch of duplicate code you should make an /obj/structure/wall_cabinet which oxygen_cabinet is a child of and make the fire extinguisher version a child as well